variables.tf

```diff
@@ -303,3 +303,24 @@ variable "cluster_autoscaler_helm_config" {
   description = "Cluster Autoscaler Helm Config"
   default     = { "version" : "9.28.0" }
 }
+
+variable "enable_ecr_pullthrough_rule" {
+  type        = bool
+  description = "Specifies if ECR pull-through rule, and accompanying resources, will be created."
+  default     = false
+}
```

Comment:
For me, this is the only variable that should be accessible from the outside. I see no use case for the two other variables `enable_ect_pullthrough_secret` and `registry_credentials`. Therefore, we should remove them from the `variables.tf` file.

Reply:
The reason why there are multiple variables:
1. enable_ecr_pullthrough_rule controls whether we want to create this ECR rule in the first place (maybe it already exists, ie. it should be done only once per account, so no need to re-do it multiple times).  
2. enable_ecr_pullthrough_secret controls whether we want to create credentials secret - if the secret is already in place (ie only rule needs creation/update), it will throw an error in TF.  
3. registry_credentials stores the actual credentials (null for our Azure registry).
So, first 2 control creation and subsequent creation/idempotency, and 3rd just stores credentials.  In my mind, only 3rd could be potentially removed, the rest is highly recommended, as found out during testing.  
Even then, 3rd var is simply best practice on how to store registry credentials (ie avoid hardcoded values).

Reply:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/secretsmanager_secret_version#key-value-pairs

Reply:
The fact that the pull-through cache rule may already exist in the AWS account is a good argument. Therefore, we should include the variable use_ecr_pullthrough_cache in the variables.tf file.
If the value of the variable is set to "true", the policy should be added to the node's role and the pull-through cache rule and the required secret should be created. Can we check if the pull-through cache rule and secret already exist and only create them if needed?
I don't see a use case for more new variables in the values.tf file. I want to keep the interface (variables.tf file) as simple as possible.